File: 프로그래머스/0/120846. 합성수 찾기/합성수 찾기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    int answer = 0;
    vector<int> result(n+1, 0);

    for (int i = 1; i <= n; i++) {
        for (int j = 1; j <= n; j++) {
            if (i * j <= n) {
                result[i * j]++;
            } else {
                break;
            }
        }
    }

    for (int i = 1; i <= n; i++) {
        if(result[i] >= 3 ) {
            answer++;
        }
    }

    return answer;
}

File: 프로그래머스/0/120846. 합성수 찾기/합성수 찾기.swift
import Foundation

func solution(_ n:Int) -> Int {
    
    var result: [Int] = Array(repeating: 0, count: n+1)
    
    for i in 1...n {
        stride(from: i, through: n, by: i).forEach { result[$0] += 1 }
    }
    
    return result.filter { $0 >= 3 }.count
}